src/main/java/org/opensearch/knn/training/TrainingJobRunner.java

```diff
@@ -95,6 +111,7 @@ public void execute(TrainingJob trainingJob, ActionListener<IndexResponse> liste
         // Serialize model before training. The model should be in the training state and the model binary should be
         // null. This notifies users that their model is training, but not yet ready for use.
         try {
+            trainingJob.getModel().getModelMetadata().setNodeAssignment(clusterService.localNode().getEphemeralId());
```

Comment:
mentioned above, I think this line wont necessarily be needed.

Reply:
Yeah I should be able to move this logic to where the TrainingJob is created